clientconn.go

```diff
@@ -152,6 +152,16 @@ func NewClient(target string, opts ...DialOption) (conn *ClientConn, err error)
 	for _, opt := range opts {
 		opt.apply(&cc.dopts)
 	}
+
+	// Determine the resolver to use.
+	if err := cc.parseTargetAndFindResolver(); err != nil {
```

Comment:
Would you mind converting this and `cc.determineAuthority` so they are not `cc` methods?  Otherwise they might forget they're not supposed to be using `cc.channelz`.

Reply:
This reads and writes a lot of stuff from cc. Unless you want me to pass in the target, dial options, and return a bunch of things to NewClient to write to cc's stuff. For the sake of readability I think it's better to keep it on cc.

Reply:
This used to all be in the function body inline and Easwar rewrote it (which I reviewed) to be on cc to factor out functionality into readable helpers and keep this function body clean.

Reply:
I guess let's rename these to `initParsedTargetAndResolverBuilder` and `initAuthority` and then at least it's obvious they're being called before the cc is fully initialized.